packages/cw-orch-mock/src/core.rs

```diff
@@ -175,6 +175,23 @@ impl<S: StateInterface> Mock<S> {
         self.state.borrow_mut().set_code_id(contract_id, code_id);
         Ok(resp)
     }
+
+    #[cfg(feature = "snapshot-testing")]
+    /// This functions allows for making sure the tests are stabilised and changes made to contracts don't have an impact on the internal storage
+    /// Usage:
+    /// ```rust,ignore
+    /// mock.take_snapshot();
+    /// ```
+    /// Find more details on how snapshot testing works on the official quick-start guide: https://insta.rs/docs/quickstart/
+    /// This function will panic if the snapshot is different from the reference snapshot
+    pub fn take_snapshot(&self) -> Result<(), CwEnvError> {
```

Comment:
Should we allow for an optional snapshot name here? 

Can be provided as the first argument to the macro:
https://docs.rs/insta/1.34.0/insta/macro.assert_yaml_snapshot.html